src/rules/noRedundantJsdocRule.ts

```diff
@@ -0,0 +1,143 @@
+        return this.applyWithFunction(sourceFile, walk);
+    }
+}
+
+function walk(ctx: Lint.WalkContext<void>): void {
+    ts.forEachChild(ctx.sourceFile, function cb(node) {
+        if (node.jsDoc !== undefined) {
+            for (const { tags } of node.jsDoc) {
+                if (tags !== undefined) {
+                    for (const tag of tags) {
+                        checkTag(tag);
+                    }
+                }
+            }
+        }
+        ts.forEachChild(node, cb);
+    });
+
+    function checkTag(tag: ts.JSDocTag): void {
+        switch (tag.kind) {
+            case ts.SyntaxKind.JSDocTag:
+                if (isRedundantTag(tag.tagName.text)) {
+                    ctx.addFailureAtNode(tag.tagName, Rule.FAILURE_STRING_REDUNDANT_TAG(tag.tagName.text));
+                }
+                break;
+
+            case ts.SyntaxKind.JSDocAugmentsTag:
+                // OK
+                break;
+
+            case ts.SyntaxKind.JSDocTemplateTag:
+            case ts.SyntaxKind.JSDocTypeTag:
+            case ts.SyntaxKind.JSDocTypedefTag:
+            case ts.SyntaxKind.JSDocPropertyTag:
+                // Always redundant
+                ctx.addFailureAtNode(tag.tagName, Rule.FAILURE_STRING_REDUNDANT_TAG(tag.tagName.text));
+                break;
+
+            case ts.SyntaxKind.JSDocReturnTag:
+            case ts.SyntaxKind.JSDocParameterTag: {
+                const { typeExpression, comment } = tag as ts.JSDocReturnTag | ts.JSDocParameterTag;
+                if (typeExpression !== undefined) {
+                    ctx.addFailureAtNode(typeExpression, Rule.FAILURE_STRING_REDUNDANT_TYPE);
+                }
+                if (comment === "") {
+                    // Redundant if no documentation
+                    ctx.addFailureAtNode(tag.tagName, Rule.FAILURE_STRING_NO_COMMENT(tag.tagName.text));
+                }
+                break;
+            }
+
+            default:
+                throw new Error(`Unexpected tag kind: ${ts.SyntaxKind[tag.kind]}`);
+        }
+    }
+}
+
+function isRedundantTag(tagName: string): boolean {
+    switch (tagName) {
+        case "abstract":
+        case "access":
+        case "class":
+        case "constant":
+        case "constructs":
+        case "default":
+        case "enum":
+        case "exports":
+        case "function":
+        case "global":
+        case "implements":
+        case "interface":
+        case "instance":
+        case "member":
+        case "memberof":
+        case "mixes":
+        case "mixin":
+        case "module":
+        case "name":
+        case "namespace":
+        case "override":
+        case "private":
+        case "property":
+        case "protected":
+        case "public":
+        case "readonly":
+        case "requires":
+        case "static":
+        case "this":
+            return true;
+        default:
+            return false;
+    }
+}
+
+declare module "typescript" {
```

Comment:
please don't do that
that should be done upstream in typescript. IIRC there's already an issue with a feature request for something like `node.getJsDoc()`